### DISCUSSION TITLE
Filebeat added inputs and port bindings

### DISCUSSION THREAD

**Discussion by Sahale** [Security-Onion-Solutions/securityonion#8463] [Ideas] [answered]
Hello!

During upgrade testing environment to .140 we faced up with error:
```
...
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 261, in render_tmpl
    output = render_str(tmplstr, context, tmplpath)
  File "/usr/lib/python3.6/site-packages/salt/utils/templates.py", line 509, in render_jinja_tmpl
    raise SaltRenderError("Jinja variable {}{}".format(exc, out), buf=tmplstr)
salt.exceptions.SaltRenderError: Jinja variable the template 'filebeat/map.jinja' (imported on line 22) does not export the requested name 'SO'
[CRITICAL] Rendering SLS 'base:filebeat' failed: Jinja variable the template 'filebeat/map.jinja' (imported on line 22) does not export the requested name 'SO'
[INFO    ] Executing command 'ls' in directory '/root'
local:
    Data failed to compile:
----------
    Rendering SLS 'base:filebeat' failed: Jinja variable the template 'filebeat/map.jinja' (imported on line 22) does not export the requested name 'SO'
...
```
It's because we use modified init.sls file for added inputs ([according to this](https://github.com/Security-Onion-Solutions/securityonion/discussions/5956))
It's the second time we stuck with this problem during update.

So maybe it's better to autobind ports that added in inputs? Or it's a bad idea?
Thanks!

**Comment by defensivedepth** [comment, marked as answer]
I added a suggestion to that other Discussion: https://github.com/Security-Onion-Solutions/securityonion/discussions/5956#discussioncomment-3328606

**Reply by Sahale** [reply]
My idea was to ask to implement autobinding for added filebeat syslog inputs like it's working for added filebeat modules now.
For example, if you add CheckPoint module and set var.source_port, SO automatically bind it so it becomes accessible from the network. But if you want to add new syslog port, than you need to modify local copy of filebeat `init.sls`.
Of course creating backup of init.sls and reapplying local configs is possible but less convenient.
Thanks!
